src/handlers/walletReadyEvents.ts

```diff
@@ -114,30 +111,37 @@ export const runFeatureUnlockChecks = async () => {
   logger.log('WALLETS TO CHECK', walletsToCheck);
 
   if (!walletsToCheck.length) return;
-
-  const featuresToUnlock = [
-    {
-      check: optimismNftAppIconCheck,
-      name: UNLOCK_KEY_OPTIMISM_NFT_APP_ICON,
-    },
-  ];
-
   const mmkv = new MMKV();
 
-  for (const feature of featuresToUnlock) {
+  let explainSheetType;
+
+  for (const iconName in unlockableAppIcons) {
```

Comment:
The way this is named, you're restricting this feature to only unlock app icons. While this is true right now, we could be unlocking other features in the future, which is why I had a generic name first.
Maybe you can still name it features, and add a `type` property = `app_icon` and wrap this logic for it.
That way we can easily add other features in the feature

Reply:
good call

i've generalized it to just 
```
  featureUnlockChecks.some(
    async checkFeature => await checkFeature(walletsToCheck)
  );
```

i think any type of unlockable feature will be able to have an unlock method that follows the pattern:
(walletsToCheck) => true if unlocked, false otherwise